multidict/_multidict_base.py

```diff
@@ -73,59 +76,43 @@ def _viewbaseset_richcmp(
 
 
 def _viewbaseset_and(
-    view: set[object], other: object
+    view: _ViewArg, other: object
 ) -> Union[set[object], NotImplementedType]:
     if not isinstance(other, Iterable):
         return NotImplemented  # type: ignore[no-any-return]
-    if isinstance(view, Set):
-        view = set(iter(view))
-    if isinstance(other, Set):
-        other = set(iter(other))
-    if not isinstance(other, Set):
-        other = set(iter(other))
-    return view & other
+    lft = set(iter(view))
+    rgt = set(iter(other))
+    return lft & rgt  # type: ignore[return-value]
```

Comment:
Why are we getting a type error here now?

Reply:
Good question.

Reply:
> got "set[Sequence[object]]"

That doesn't sound right to me..

Reply:
Because it is applicable to both KeysView and ItemsView.
ItemsView returns `tuple[str, T]`.

Honestly, I don't think that this file should have type annotations.
It is never called from python but from C Extension.
mypy check is useless here; it doesn't analyze the whole path, and it has no idea what types were used by C Extension during the call.

Reply:
OK, KeysView[str] is going to make an iterator of strs, while ItemsView[str, object] is going to make an iterator of tuple[str, object]. The common supertype of those is Iterator[Sequence[object]] (as str is a sequence itself).